src/sync.ts

```diff
@@ -1264,6 +1268,28 @@ export class SyncApi {
                 }
             }
 
+            room.resetThreadUnreadNotificationCount();
+            const unreadThreadNotifications = joinObj[UNREAD_THREAD_NOTIFICATIONS.name];
+            if (unreadThreadNotifications) {
+                Object.entries(unreadThreadNotifications).forEach(([threadId, unreadNotification]) => {
+                    room.setThreadUnreadNotificationCount(
+                        threadId,
+                        NotificationCountType.Total,
+                        unreadNotification.notification_count,
+                    );
+
+                    const hasUnreadNotification =
+                        room.getThreadUnreadNotificationCount(threadId, NotificationCountType.Highlight) <= 0;
```

Comment:
Is `<= 0` correct here? I would expect to check for `> 0`.

Reply:
I was confused when I named that variable. I replicated the logic from above
https://github.com/matrix-org/matrix-js-sdk/blob/7d5adf48bcca8fcdcefea707528aa906c71929e2/src/sync.ts#L1263